synapse/tests/test_lib_ast.py

```diff
@@ -651,6 +700,204 @@ class AstTest(s_test.SynTest):
+            $lib.print("biz is now {biz}", biz=$biz)
+            '''
+            msgs = await core.streamstorm(q).list()
+            prints = list(filter(lambda m: m[0] == 'print', msgs))
+            self.len(6, prints)
+            self.stormIsInPrint("neato burrito", msgs)
+            self.stormIsInPrint("done", msgs)
+            self.stormIsInPrint("doublewoot", msgs)
+            self.stormIsInPrint("biz is now 20", msgs)
+
+            # test that we the functions in a module don't pollute our own runts
+            q = '''
+            $test=$lib.import(test)
+            $lib.print($outer("1337"))
+            '''
+            msgs = await core.streamstorm(q).list()
+            for msg in msgs:
+                self.ne('print', msg[0])
+
+            # make sure we can set variables to the results of other functions in the same query
+            q = '''
+            function baz(arg1) {
+                $lib.print('arg1={a}', a=$arg1)
+                return ($arg1)
+            }
+            function bar(arg2) {
+                $lib.print('arg2={a}', a=$arg2)
+                $retn = $baz($arg2)
+                return ($retn)
+            }
+            $foo = $bar("hehe")
+            $lib.print($foo)
+            '''
+            msgs = await core.streamstorm(q).list()
+            self.stormIsInPrint('hehe', msgs)
+            self.stormIsInPrint('arg1=hehe', msgs)
+            self.stormIsInPrint('arg2=hehe', msgs)
+
+            # call an import and have it's module local variables be mapped in to its own scope
+            q = '''
+            $test = $lib.import(importnest)
+            $haha = $test.outer(False, $(33))
+            $lib.print($haha)
+            $hehe = $test.outer(True, $(17))
+            $lib.print($hehe)
+            $retn = $lib.import(importnest).outer(True, $(90))
+            $lib.print($retn)
+            $lib.print("counter is {c}", c$counter)
+            '''
+            msgs = await core.streamstorm(q).list()
+            prints = list(filter(lambda m: m[0] == 'print', msgs))
+            self.len(9, prints)
+            self.stormIsInPrint('counter is 0', msgs)
+            self.stormIsInPrint('foobar is 33', msgs)
+            self.stormIsInPrint('(Run: 0) we got back bar', msgs)
+            self.stormIsInPrint('counter is 33', msgs)
+            self.stormIsInPrint('foobar is 50', msgs)
+            self.stormIsInPrint('(Run: 33) we got back foo', msgs)
+            self.stormIsInPrint('counter is 0', msgs)
+            self.stormIsInPrint('foobar is 90', msgs)
+            self.stormIsInPrint('(Run: 0) we got back foo', msgs)
+
+            # yields all the way down, no imports
+            q = '''
+            $count = 0
+            function baz(arg3) {
+                [ test:str = $arg3 ]
+                $count = $( $count + 1)
+                [ test:str = "cool" ]
+            }
+
+            function bar(arg2) {
+                yield $baz($arg2)
```

Comment:
I thought we gutted the yield keyword from the function syntax?

Reply:
Maybe it's still in PR? Because this syntax still yields out the node I expect.